vehicle/Car.h

```diff
@@ -10,20 +10,20 @@
 class Car {
 public:
     Car(std::unique_ptr<IEngine> p_engine, std::unique_ptr<IGear> p_gear)
-    : p_engine(std::move(p_engine)), p_gear(std::move(p_gear)) {}
+    : m_engine(std::move(p_engine)), m_gear(std::move(p_gear)) {}
 
     virtual std::string getCarType() const = 0;
     const IEngine *getEngine() const {
-        return p_engine.get();
+        return m_engine.get();
     }
     const IGear *getGear() const {
-        return p_gear.get();
+        return m_gear.get();
     }
     virtual ~Car() {}
 
 protected:
-    std::unique_ptr<IEngine> p_engine;
-    std::unique_ptr<IGear> p_gear;
+    std::unique_ptr<IEngine> m_engine;
```

Comment:
here's the trick how you can reduce extra long names: add something like 
using Engine = std::unique_ptr<IEngine>;

then everywhere this line is included, you can simply type "Engine" instead of the full one